dto.go

```diff
@@ -0,0 +1,75 @@
+package basm
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/mailru/easyjson"
+)
+
+func marshal(v any) ([]byte, error) {
+	easy, ok := v.(easyjson.Marshaler)
+	if !ok {
+		return nil, errors.New("value does not implement easyjson.Marshaler")
+	}
+	return easyjson.Marshal(easy)
+}
+
+func unmarshal(data []byte, v any) error {
+	easy, ok := v.(easyjson.Unmarshaler)
+	if !ok {
+		return errors.New("value does not implement easyjson.Unmarshaler")
+	}
+	return easyjson.Unmarshal(data, easy)
+}
+
+type httpRequestInput struct {
+	Method  string              `json:"method"`
+	URL     string              `json:"url"`
+	Headers map[string][]string `json:"headers"`
+	Body    []byte              `json:"body"`
+}
+
+type httpRequestOutput struct {
+	StatusCode int                 `json:"status_code"`
+	Body       []byte              `json:"body"`
+	Headers    map[string][]string `json:"headers"`
+}
+
+type verifyAttestationInput struct {
+	EnclaveAttestedKey    []byte `json:"enclave_attested_app_public_key"`
+	TransitiveAttestation []byte `json:"transitive_attestation"`
+	AcceptableMeasures    []byte `json:"acceptable_measurements"`
+}
+
+type verifyAttestationOutput struct {
+	RawClaims []byte `json:"raw_claims"`
+}
+
+type result struct {
+	IsOK  bool            `json:"ok"`
+	Error string          `json:"error"`
+	Value json.RawMessage `json:"value"`
+}
+
+func readHostResult[T any](data []byte) (T, error) {
```

Comment:
the signature here makes it so the type needs to be specified when using the function. We could update to something like 
`readHostResult(T any)(data []byte, v *T) error` if we want to have a usage pattern more like
```go
var output httpRequestOutput
	err = readHostResult(resultData, &output)
	if err != nil {
		msg := "reading host fn result: " + err.Error()
		return httpRequestOutput{}, errors.New(msg)
	}
	return output, nil
```

Reply:
updated to follow the marshal/unmarshal pattern